### PR TITLE
Made operation timeout for memcached memdatacache configuration driven,

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
Disabled blocking call on caching future.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6250795&bid=20355178)

**Review by bharatmotwani — APPROVED** [review]

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    /**
+     * Create a BinaryConnectionFactory with the given maximum operation queue length, and the given read buffer size
+     * and timeout in milliseconds.
+     *
+     * @param len  the length of the queue
+     * @param bufSize the size of the queue
+     * @param operationTimeout  The number of milliseconds to wait for timeout,
+     */
+    TimeoutConfigurerBinaryConnectionFactory(int len, int bufSize, Long operationTimeout) {
```
Comment (archolewa) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java:40]: These constructors are intentionally package private?

**Review by archolewa — APPROVED** [review]

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java (hunk 1 above)
Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java:40]: Nope.

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6250795&bid=20355178)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/TimeoutConfigurerBinaryConnectionFactory.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6250795&bid=20355178)

**Pull request merged**